javascript/javascript1/week2/smartease.js

```diff
@@ -0,0 +1,177 @@
+
+//Formal fullname
+console.log("**********************");
+console.log("Ex 2: Formal fullname");
+console.log("**********************");
+function getFormalFullName(firstname, surname, useFormalName, gender) {
+  if (!firstname || !surname) {
+    return "Please provide your firstname and surname";
+  } else {
+    if (useFormalName === false || !useFormalName) {
+      return `${firstname} ${surname}`;
+    } else if (typeof useFormalName !== "boolean") {
+      return "please mention useFormalName as true or false";
+    } else if (useFormalName === true && gender === "male") {
+      return `Lord ${firstname} ${surname}`;
+    } else if (useFormalName === true && gender === "female") {
+      return `Lady ${firstname} ${surname}`;
+    } else {
+      return "Please provide the gender";
+    }
+  }
+}
+const formalFullName1 = getFormalFullName("Benjamin", "Hughes", false);
+const formalFullName2 = getFormalFullName("Ida", "Jones");
+console.log(
+  `formalFullName1:${formalFullName1}\nformalFullName2:${formalFullName2}`
+);
+
+//Event application
+console.log("**********************");
+console.log("Ex 3: Event application");
+console.log("**********************");
+
+function getEventWeekday(daysAfter) {
+  const today = new Date().getDay();
+  const weekDays = [
+    "Sunday",
+    "Monday",
+    "Tuesday",
+    "Wednesday",
+    "Thursday",
+    "Friday",
+    "Saturday",
+  ];
+
+  return weekDays[(today + daysAfter) % 7];
+}
+const daysAfter = 4;
+const eventDay = getEventWeekday(daysAfter);
+console.log(eventDay);
+
+//Weather wear
+console.log("**********************");
+console.log("Ex 4: Weather wear");
+console.log("**********************");
+
+function selectClothes(temperature) {
+  if (temperature <= 5) {
+    return "Winter Jacket and Thermal Wear";
+  } else if (temperature > 5 && temperature < 18) {
+    return "Normal Jacket and Pant and T-shirt";
+  } else if (temperature >= 18) {
+    return "Shorts and a T-shirt";
+  } else {
+    return "Please provide the current temperature in degrees";
+  }
+}
+const temperature = 18;
+const clothesToWear = selectClothes(temperature);
+console.log(clothesToWear);
+
+//Student manager
+console.log("**********************");
+console.log("Ex 5: Student manager");
+console.log("**********************");
+let class07Students = [];
+function addStudentToClass(studentName) {
+  studentName = studentName.toLowerCase();
+  if (
+    (class07Students.length < 6 &&
+      !class07Students.includes(studentName) &&
+      studentName !== "") ||
+    studentName === "mary elizabeth"
+  ) {
+    class07Students.push(studentName);
+  } else if (studentName === "") {
+    console.log("Please provide the student name");
+  } else if (class07Students.includes(studentName)) {
+    console.log(`Student ${studentName} is already in the class`);
+  } else {
+    console.log("Cannot add more students to class 07");
+  }
+  return class07Students;
+}
```

Comment:
Nice  this works! Yet, I would have checked all the negatives in `if` and `else if` first, and in the final `else` actually `push` to the array. Right now it is hard to understand what you are checking where.

Reply:
Yes, you are correct. I realized that organizing the negatives in the if and else if statements first, and pushing to the array in the final else statement, provides better clarity. I have made the necessary changes accordingly. Thank you for the suggestion!